week2/homework/src/index.js

```diff
@@ -1,3 +1,99 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const fs = require('fs');
+
+const DEFAULT_ENCODING = 'utf8';
+const STORE_FILE_NAME  = 'store.txt';
+
+function printHelp() {
+  console.log(`Usage: node index.js [options]
+
+  - HomeWork To-Do App  -
+
+Options:
+
+  list                     show all to-dos
+  add    [to-do]           add to-do
+  remove [number of to-do] remove to-do
+  reset                    remove all to-dos
+  help                     show this help text
+  `);
+}
+
+const [,, cmd, ...args] = process.argv;
+
+switch (cmd) {
+  case 'add':
+    const todo = args.join(' ');
+    
+    fs.appendFile(STORE_FILE_NAME, `${todo}\n`, (err) => {
+    console.error(err);    
+    });
+    
+    console.log(`"${todo}" to-do added`);
+  break;
+
+  case 'remove':
+    fs.readFile(STORE_FILE_NAME, DEFAULT_ENCODING, (err, data) => {
+      if (err) {
+        console.error(err);
+      }
+      const removeTodoNumber = Number(args.toString());
+      if (isNaN(removeTodoNumber)) {
+        console.error(`to remove type to-do's number`);
+      }
+      else {
+        const todoList = data.split('\n');
+        todoList.pop();
+        const removeTodoIndex = removeTodoNumber - 1;
+        todoList.splice(removeTodoIndex, 1);
+        
+        fs.unlink(STORE_FILE_NAME, (err) => {
+          if (err) {
+            console.error(err);
+          }
+        });
+        
+        todoList.forEach(todo => {
```

Comment:
@Ann-Grabetski I am afraid this code has a subtle but serious flaw. Both `unlink` and `appendFile` are async function, which means there is no guarantee that unlink`` will finish before `appendFile` or that the `appendFile` will be execute in the correct order. You should either use the synchronous versions or take care to ensure the correct order.